packages/server/src/handlers/BaseHandler.ts

```diff
@@ -101,4 +110,58 @@ export class BaseHandler extends EventEmitter {
 
     return {host: host as string, proto}
   }
+
+  protected getLocker(req: http.IncomingMessage) {
+    if (typeof this.options.locker === 'function') {
+      return this.options.locker(req)
```

Comment:
Should we allow `this.options.locker` to be an `async` function? People will likely ask for this to do asynchronous lookups, for example.

Reply:
Do you have a use case in mind in which it could be beneficial to have it async?
I can't think of one atm

Reply:
Done